source/slang/slang-compiler.cpp

```diff
@@ -2177,6 +2177,103 @@ namespace Slang
         return SLANG_OK;
     }
 
+    static void writeString(Stream& stream, const char* string)
+    {
+        stream.write(string, strlen(string));
+    }
+
+    static void escapeDependencyString(const char* string, StringBuilder& builder)
+    {
+        // make has unusual escaping rules, but we only care about characters that are acceptable in a path
+        for (const char* p = string; *p; ++p)
+        {
+            char c = *p;
+            switch(c)
+            {
+            case ' ':
+            case ':':
+            case '#':
+            case '[':
+            case ']':
+            case '\\':
+                builder.appendChar('\\');
+                break;
+
+            case '$':
+                builder.appendChar('$');
+                break;
+            }
```

Comment:
Is the '$' case necessary?

Reply:
Yes: per Makefile conventions, `$` is escaped as `$$`, unlike other characters.

Reply:
Ok makes sense.